vault/seal/transit/transit.go

```diff
@@ -0,0 +1,256 @@
+	}
+	if disableRenewalRaw != "" {
+		var err error
+		disableRenewal, err = strconv.ParseBool(disableRenewalRaw)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	var namespace string
+	switch {
+	case os.Getenv("VAULT_NAMESPACE") != "":
+		namespace = os.Getenv("VAULT_NAMESPACE")
+	case config["namespace"] != "":
+		namespace = config["namespace"]
+	}
+
+	apiConfig := api.DefaultConfig()
+	if config["address"] != "" {
+		apiConfig.Address = config["address"]
+	}
+	if config["tls_ca_cert"] != "" || config["tls_ca_path"] != "" || config["tls_client_cert"] != "" || config["tls_client_key"] != "" ||
+		config["tls_server_name"] != "" || config["tls_skip_veriry"] != "" {
+		var tlsSkipVerify bool
+		if config["tls_skip_verify"] != "" {
+			var err error
+			tlsSkipVerify, err = strconv.ParseBool(config["tls_skip_verify"])
+			if err != nil {
+				return nil, err
+			}
+		}
+
+		tlsConfig := &api.TLSConfig{
+			CACert:        config["tls_ca_cert"],
+			CAPath:        config["tls_ca_path"],
+			ClientCert:    config["tls_client_cert"],
+			ClientKey:     config["tls_client_key"],
+			TLSServerName: config["tls_server_name"],
+			Insecure:      tlsSkipVerify,
+		}
+		if err := apiConfig.ConfigureTLS(tlsConfig); err != nil {
+			return nil, err
+		}
+	}
+
+	if s.client == nil {
+		client, err := api.NewClient(apiConfig)
+		if err != nil {
+			return nil, err
+		}
+		if config["token"] != "" {
+			client.SetToken(config["token"])
+		}
+		if namespace != "" {
+			client.SetNamespace(namespace)
+		}
+		if client.Token() == "" {
+			return nil, errors.New("missing token")
+		}
+		s.client = client
+
+		// Send a value to test the seal and to set the current key id
+		if _, err := s.Encrypt(context.Background(), []byte("a")); err != nil {
+			return nil, err
+		}
+
+		if !disableRenewal {
+			// Renew the token immediately to get a secret to pass to renewer
+			secret, err := client.Auth().Token().RenewTokenAsSelf(s.client.Token(), 0)
+			// If we don't get an error renewing, set up a renewer.  The token may not be renewable or not have
+			// permission to renew-self.
+			if err == nil {
+				renewer, err := s.client.NewRenewer(&api.RenewerInput{
+					Secret: secret,
+				})
+				if err != nil {
+					return nil, err
+				}
+				s.renewer = renewer
+
+				go func() {
+					for {
+						select {
+						case err := <-renewer.DoneCh():
+							s.logger.Info("shutting down token renewal")
+							if err != nil {
+								s.logger.Error("error renewing token", "error", err)
+							}
+							return
+						case <-renewer.RenewCh():
+							s.logger.Trace("successfully renewed token")
+						}
+					}
+				}()
+				go s.renewer.Renew()
+			} else {
+				s.logger.Info("unable to renew token, disabling renewal", "err", err)
+			}
+		}
+	}
+
+	sealInfo := make(map[string]string)
+	sealInfo["address"] = s.client.Address()
+	sealInfo["mount_path"] = s.mountPath
+	sealInfo["key_name"] = s.keyName
+	if namespace != "" {
+		sealInfo["namespace"] = namespace
+	}
+
```

Comment:
I think we should query the API to get the latest key term here and set the `currentKeyID` value. Probably best to just encrypt some dummy data so the ACL doesn't also need key read access.